### PR TITLE
security: update go version in asdf to 1.18.7

### PR DESCRIPTION
**Pull request by burmudar**
Update go version to latest security release see https://groups.google.com/g/golang-announce/c/xtuG5faxtaU
## Test plan
Green CI

### CODE REVIEW

File: .tool-versions [hunk 1]
```diff
@@ -1,4 +1,4 @@
-golang 1.18.1
+golang 1.18.7
```
Comment (jhchabran) [inline .tool-versions:1]: We need to update the check in `sg` as well :) 

https://github.com/sourcegraph/sourcegraph/commit/cc3010911fb4093c71236a78cd72f7767630e92d#diff-d6971f9b789dfaad28401bbb466beee887d64bb586d8bb9e2bc0567136e5cdbdR38

**Review by jhchabran — CHANGES_REQUESTED** [review]
Good call to update this 🙏

File: .tool-versions (hunk 1 above)
Comment (burmudar) [reply .tool-versions:1]: Good spot! Lemme update that 🚒

**Comment by burmudar** [comment]
For future reference - I tried updating to `1.19.7` and there were about 3 test failures which don't seem like a lot of work to update https://buildkite.com/sourcegraph/sourcegraph/builds/175939#0183a76f-d17a-4eff-9f2a-4cc227881969

**Review by jhchabran — APPROVED** [review]

**Pull request merged**